src/util/simplify_expr2.cpp

```diff
@@ -1289,7 +1289,25 @@ static expr2tc do_bit_munge_operation(
   if(is_signedbv_type(type))
     theint = new constant_int2t(type, BigInt(val1));
   else
-    theint = new constant_int2t(type, BigInt((uint64_t)val1));
+  {
+    switch(type->get_width())
+    {
+    case 8:
+      theint = new constant_int2t(type, BigInt((uint8_t)val1));
+      break;
+    case 16:
+      theint = new constant_int2t(type, BigInt((uint16_t)val1));
+      break;
+    case 32:
+      theint = new constant_int2t(type, BigInt((uint32_t)val1));
+      break;
+    case 64:
+      theint = new constant_int2t(type, BigInt((uint64_t)val1));
+      break;
+    default:
+      abort();
+    }
+  }
```

Comment:
shouldn't we do that for the signedbv case as well?

Reply:
Fixed.